### PR TITLE
Support partially matching ID

### PR DESCRIPTION
**Pull request by deiwin**

### CODE REVIEW

File: lib/pains.js [hunk 1]
```diff
@@ -4,10 +4,17 @@ var Promise = require('rsvp').Promise;
 module.exports = function(addon) {
   var PAINS_KEY = 'pains';
 
+  var escapeRegExp = function(str) {
+    return str.replace(/[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, "\\$&");
```
Comment (urmastalimaa) [inline lib/pains.js]: wth is this?

Comment (deiwin) [inline lib/pains.js]: This is how backdoors are created ;)

Forgot to link to source. https://github.com/salemove/akahsa/pull/1/commits/d78fe592c3858a56d8e772b1446ef24d11e5f32a

File: test/pains_test.js [hunk 2]
```diff
@@ -198,6 +198,15 @@ describe('pains', function() {
 
         itUpdatesTheDbRecord();
       });
+
+      context('with the description partially matching the previous ID', function() {
+        pain.is(R.always({
+          id: 'another id',
+          description: previousId.substring(0, 8),
```
Comment (urmastalimaa) [inline test/pains_test.js]: shouldn't this be previousDescription?

Comment (deiwin) [inline test/pains_test.js]: No, this is correct. The `description` here is basically everything that follows the `/ouch` command (or `/heal` in the other added tests). So this test basically tests that if one were to write `/ouch d9872979` the pain with ID `d9872979-d66a-41a7-a83b-31f13182f9ae` should get reported. I understand the confusion and want to change it, but I'd rather do it separately. I think `pain` in this context could be renamed to something like `command` and `pain.description` of the input to something like `command.text`.

**Comment by urmastalimaa** [comment]
+1

**Pull request merged**